src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -133,10 +134,13 @@ else if (!savedSenderID.equals(senderID)) {
 
                     }
 
-                    if (gCachedExtras != null) {
+                    if (!gCachedExtras.isEmpty()) {
                         Log.v(LOG_TAG, "sending cached extras");
-                        sendExtras(gCachedExtras);
-                        gCachedExtras = null;
+                        Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
```

Comment:
This is where we could get into a problem if a new push message comes in while iterating over the list. So we need to access the iterator in a synchronized block:

```
synchronized(gCachedExtras) {
    Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
    while (gCachedExtrasIterator.hasNext()) {
        sendExtras(gCachedExtrasIterator.next());
    }
}
```